api/v1/helpers/verbs/utils.js

```diff
@@ -41,13 +41,11 @@ function updateInstance(o, puttableFields, toPut) {
       }
 
       o.set(key, nullish);
-
-      // take nullified fields out of changed fields
-      o.changed(key, false);
```

Comment:
This is the line that breaks the PUT behavior. The DB will only update fields with changed=true, so setting it to false for missing fields means fields that are left blank in the request will not be updated to null/false. This makes the PUT behave like a PATCH, only updating the fields that were specified and ignoring all others.

This line was added about a year ago. I think it was added to get tests passing without realizing the implications.

Can you confirm this is not the intended behavior?

PUT has been acting like PATCH for the last year. We should make sure no one is relying on this before we change it back. (I believe the original bug that triggered this investigation was caused by a request that was relying on PUT acting like a PATCH)

Reply:
@jgraff2 FTW!!!!!! THIS IS HUGE!!!!
Almost all the PUTs in our logs from the last 30 days are to /v1/aspects, from the integration with the most "sample missing name" problems. @shriramshankar 
Just make sure this doesn't break anything with PUT/PATCH lens and bot (which are different cuz of file upload).

Reply:
Does it mean this was causing the "sample with missing name" problem?

Reply:
I'm not sure I can claim that with any confidence, but it is surely related.

Reply:
PUT for lens and bots doesn't use doPut, or call updateInstance directly, so this won't have any effect on those.